server/controllers/api/claim/update/index.js

```diff
@@ -0,0 +1,151 @@
+const parsePublishApiRequestBody = require('../publish/parsePublishApiRequestBody');
+const {parsePublishApiRequestFiles, parsePublishApiRequestThumbnail} = require('../publish/parsePublishApiRequestFiles.js');
+const authenticateUser = require('../publish/authentication.js');
+const createThumbnailPublishParams = require('../publish/createThumbnailPublishParams.js');
+
+/*
+  route to update a claim through the daemon
+*/
+
+const updateMetadata = ({nsfw, license, title, description}) => {
+  const update = {};
+  if (nsfw) update['nsfw'] = nsfw;
+  if (license) update['license'] = license;
+  if (title) update['title'] = title;
+  if (description) update['description'] = description;
+  return update;
+};
+
+const claimUpdate = ({ body, files, headers, ip, originalUrl, user, tor }, res) => {
+  // logging
+  logger.info('Claim update request:', {
+    ip,
+    headers,
+    body,
+    files,
+    user,
+  });
+
+  // check for disabled publishing
+  if (disabled) {
+    return res.status(503).json({
+      success: false,
+      message: disabledMessage,
+    });
+  }
+
+  // define variables
+  let channelName;
+  let channelId;
+  let channelPassword;
+  let description;
+  let fileName;
+  let filePath;
+  let fileType;
+  let gaStartTime;
+  let thumbnail;
+  let fileExtension;
+  let license;
+  let name;
+  let nsfw;
+  let thumbnailFileName;
+  let thumbnailFilePath;
+  let thumbnailFileType;
+  let title;
+  let claimRecord;
+  let metadata;
+  // record the start time of the request
+  gaStartTime = Date.now();
+
+  try {
+    ({name, nsfw, license, title, description, thumbnail} = parsePublishApiRequestBody(body));
+    if (files.file) {
+      ({fileName, filePath, fileExtension, fileType} = parsePublishApiRequestFiles(files));
+      if (files.thumbnail) {
+        ({thumbnailFileName, thumbnailFilePath, thumbnailFileType} = parsePublishApiRequestThumbnail(files));
+      }
+    }
+    ({channelName, channelId, channelPassword} = body);
+  } catch (error) {
+    return res.status(400).json({success: false, message: error.message});
+  }
+
+  // check channel authorization
+  authenticateUser(channelName, channelId, channelPassword, user)
+    .then(({ channelName, channelClaimId }) => {
+      return db.Claim.findOne({
+        where: {
+          name,
+          channelName,
+        },
+      });
+    })
+    .then(claim => {
+      return resolveUri(`${claim.name}#${claim.claimId}`);
+    })
+    .then(fullClaim => {
+      claimRecord = fullClaim;
+      logger.info('fullClaim', fullClaim);
+      metadata = Object.assign({}, {
+        title      : claimRecord.title,
+        description: claimRecord.description,
+        nsfw       : claimRecord.nsfw,
+        license    : claimRecord.license,
+        language   : 'en',
+        author     : details.title,
+      }, updateMetadata({title, description, nsfw, license}));
+      const publishParams = {
+        name,
+        bid          : 0.01,
```

Comment:
Can be filed, but this is a rather important value that ought to be hoisted all the way to a site-specific config

Reply:
#609